docs/adr/0005-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,106 @@
+
+## Considered Options
+
+* **(A) Keep our current release process**
+* **(B) Move to a rapid release process where changes in app-services are frequently merged into consumer applications**
+    * Phase 1: update our branching model and release process (Jan 2023 - Feb 2023)
+        * Update our branches to match how firefox-android works: a `main` branch that is used for the current nightly builds and a `releases_v[N]` for each previous release.
+        * Automate a nightly release of our `main` branch.
+          * Create a commit that updates our version number to something like `[firefox-version].0-nightly-[YYYYMMDD]`, similar to what `prepare-release.py` does now.
+            * Make sure that the github CI can push this commit without needing PR approval.
+            * Note: we will no longer be using semver-style versions and also not following semver semantics.  We will be releasing breaking changes in a nightly cycle without bumping the minor version.
+          * Create a GitHub release from the branch with a tag based on the version number / date (`v[firefox-version].0-nightly-[YYYYMMDD]`). This will cause maven packages to be published, start the `rust-components-swift` CI, etc.
+        * Manually handle releases from the `release_v*` branches whenever new code is committed to those branches
+          * Update our version number to `[firefox-version].[N+1]`
+          * Create a GitHub release from the branch
+          * Create PRs to merge the changes forward to `main` and more recent release branches.
+        * Continue to maintain our CHANGELOG, but with new versioning.  In main, keep a rolling log of updates with the header `[firefox-version].0`
+        * This replaces our current release process -- no more cutting releases.
+        * To push changes to a consumer: create a PR in their repository that bumps the application-services version to an existing release.
+        * On the monthly Firefox merge date:
+          * Create a `release_v[N]` branch of of main
+          * Create a GitHub release for that branch
+          * Bump the major version number in main and start a new changelog header.
+    * Phase 2: Automate nightly merges of `main` into consumer applications (March 2023 - May 2023)
+      * Whenever a new application-services version is released (including nightly, beta, release, and ESR versions).
+        * For each consumer application (including all mobile apps and Desktop):
+            * Create a PR to take the new application-services version.
+            * Set up an auto-merge system where if the test suite passes we automatically merge the PR.
+            * Set up an alert that gets sent to `sync-bots` if the test suite fails.  This means we would need to identify why the tests failed and assign a developer to resolve the issue.  The expectation would be that the developer fixes things in time for the next nightly merge.
```

Comment:
I think there's already a similar process in place for mobile - when we make a new (manual) release, some process kicks in - @rvandermeulen understands this better than I do - maybe link to that as a point of context, just to make it clear there is already something we can beg/borrow/steal for these steps?

Reply:
For iOS we have a process in place, we check (currently everyday) if there is a new rust-components release. If there is a new one, a [PR](https://github.com/mozilla-mobile/firefox-ios/pull/12649) is created automatically. It pre-builds the app on our CI system (Bitrise) using that new version. We know then if the app will be green or not with that version. Once the job finishes, we send a [slack notification](https://mozilla.slack.com/archives/C4DK7829K/p1670598825686639) the the firefox-ios team